crates/trie/common/src/account.rs

```diff
@@ -9,21 +9,32 @@ use revm_primitives::AccountInfo;
 
 /// An Ethereum account as represented in the trie.
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Default, RlpEncodable, RlpDecodable)]
-pub struct TrieAccount {
-    /// Account nonce.
-    pub nonce: u64,
-    /// Account balance.
-    pub balance: U256,
-    /// Account's storage root.
-    pub storage_root: B256,
-    /// Hash of the account's bytecode.
-    pub code_hash: B256,
-}
+pub struct TrieAccount(AlloyTrieAccount);
```

Comment:
can we instead use the Alloy struct directly?

Reply:
I tried this but there are some foreign impls implemented on this type which are violated when used Alloy struct instead.